zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -329,7 +329,7 @@ cdef _array_for_column_impl(object dtype,
                 # upper bound doesn't include the timestamp because we've
                 # already included the timestamp-date in the baseline.
                 end = max(ts_ix - 1, 0)
-                if end >= asof_ix:
+                if end >= asof_ix and out_array[asof_ix, column_ix] != value:
```

Comment:
Also, this is technically unsafe for the special case of "baseline = N, event arrives that amends baseline to N +1, then event arrives that amends the baseline **back to N**, right?  We're just deciding that the performance benefit here is worth losing that second amendment?